File: output/gpt_4o/avatar/algo_based_translation/java/python/successful_translation/codeforces_665_A.py

import sys

def main():
    input = sys.stdin.read
    data = input().splitlines()
    
    # Read first line of input
    a, tA = map(int, data[0].split())
    
    # Read second line of input
    b, tB = map(int, data[1].split())
    
    # Read timing input
    timing = data[2]
    hrs = int(timing[0:2])
    mins = int(timing[3:])
    
    # Calculate simulated departure and arrival times
    simDeparture = hrs * 60 + mins
    simArrival = simDeparture + tA
    
    # Initialize counter
    counter = 0
    
    # Loop through each minute from 300 to 1439
    for i in range(300, 1440, b):
        busDeparture = i
        busArrival = i + tB
        if busDeparture >= simArrival or simDeparture >= busArrival:
            continue
        counter += 1
    
    # Output result
    sys.stdout.write(str(counter) + '\n')

if __name__ == "__main__":
    main()

File: output/gpt_4o/avatar/algo_based_translation/python/java/successful_translation/codeforces_665_A.java

import java.util.Scanner;

public class codeforces_665_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read and parse first line of input
        int a = scanner.nextInt();
        int at = scanner.nextInt();
        
        // Read and parse second line of input
        int b = scanner.nextInt();
        int bt = scanner.nextInt();
        
        // Read and parse third line of input (time in "HH:MM" format)
        String[] time = scanner.next().split(":");
        int t1 = Integer.parseInt(time[0]);
        int t2 = Integer.parseInt(time[1]);

        // Calculate st and fin
        int st = t2 + (t1 - 5) * 60;
        int fin = st + at;
        
        // Initialize now and ans
        int now = 0;
        int ans = 0;
        
        // Loop to count how many times task b can start after st
        while (now < fin && now < 1140) {
            if (now + bt > st) {
                ans++;
            }
            now += b;
        }
        
        // Output the result
        System.out.println(ans);
        
        scanner.close();
    }
}

